src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedInlineArrayTypeSymbol.cs

```diff
@@ -0,0 +1,240 @@
+
+        public override Accessibility DeclaredAccessibility => Accessibility.Internal;
+
+        public override bool IsSerializable => false;
+
+        public override bool AreLocalsZeroed => true; // PROTOTYPE: Is this correct?
+
+        public override TypeKind TypeKind => TypeKind.Struct;
+
+        public override bool IsRefLikeType => false;
+
+        public override bool IsReadOnly => true;
+
+        public override Symbol? ContainingSymbol => null;
+
+        internal override ModuleSymbol ContainingModule => _containingModule;
+
+        public override AssemblySymbol ContainingAssembly => _containingModule.ContainingAssembly;
+
+        public override NamespaceSymbol ContainingNamespace => _containingModule.GlobalNamespace;
+
+        public override ImmutableArray<Location> Locations => ImmutableArray<Location>.Empty;
+
+        public override ImmutableArray<SyntaxReference> DeclaringSyntaxReferences => throw new NotImplementedException();
+
+        public override bool IsStatic => false;
+
+        public override bool IsAbstract => false;
+
+        public override bool IsSealed => true;
+
+        internal override ImmutableArray<TypeWithAnnotations> TypeArgumentsWithAnnotationsNoUseSiteDiagnostics => GetTypeParametersAsTypeArguments();
+
+        internal override bool IsFileLocal => false;
+
+        internal override FileIdentifier? AssociatedFileIdentifier => null;
+
+        internal override bool MangleName => false; // PROTOTYPE: Is this correct?
+
+        internal override bool HasDeclaredRequiredMembers => false;
+
+        internal override bool HasCodeAnalysisEmbeddedAttribute => false;
+
+        internal override bool IsInterpolatedStringHandlerType => false;
+
+        internal override bool HasSpecialName => false;
+
+        internal override bool IsComImport => false;
+
+        internal override bool IsWindowsRuntimeImport => false;
+
+        internal override bool ShouldAddWinRTMembers => false;
+
+        internal override TypeLayout Layout => default; // PROTOTYPE: Is this correct?
+
+        internal override CharSet MarshallingCharSet => DefaultMarshallingCharSet;
+
+        internal override bool HasDeclarativeSecurity => false;
+
+        internal override bool IsInterface => false;
+
+        internal override NamedTypeSymbol? NativeIntegerUnderlyingType => null;
+
+        internal override NamedTypeSymbol BaseTypeNoUseSiteDiagnostics => ContainingAssembly.GetSpecialType(SpecialType.System_ValueType);
+
+        internal override bool IsRecord => false;
+
+        internal override bool IsRecordStruct => false;
+
+        internal override ObsoleteAttributeData? ObsoleteAttributeData => null;
+
+        public override ImmutableArray<Symbol> GetMembers() => ImmutableArray<Symbol>.Empty;
+
+        public override ImmutableArray<Symbol> GetMembers(string name) => GetMembers().WhereAsArray(m => m.Name == name);
```

Comment:
Could be specialized to just return the name of the field

Reply:
The current implementation seems more robust, and `WhereAsArray()` should avoid any allocations since the result should be either the entire array or an empty array.